torch/csrc/jit/mobile/import.cpp

```diff
@@ -520,10 +520,32 @@ c10::IValue BytecodeDeserializer::readArchive(
       return obj;
     }
   };
-
+  const std::string slash = "/";
   auto read_record = [&](const std::string& name) {
+    std::size_t found = name.find(slash);
     std::stringstream ss;
-    ss << archive_name << "/" << name;
+    // In version 4, the tensor root_key doesn't include the parent path
+    // To support backward compatibility, when the name doesn't include slash
+    // assume it's version 4 and attach the archive_name_plus_slash
+    // The example tensor format is:
+    // torch._utils._rebuild_tensor_v2(
+    //     pers.obj(('storage', torch.FloatStorage, '17', 'cpu', 22736),),
+    //     0,
+    //     (1, 464, 7, 7),
+    //     (22736, 49, 7, 1),
+    //     False,
+    //     collections.OrderedDict())
+    if (found == std::string::npos) {
+      ss << archive_name << slash << name;
+      return std::get<0>(reader_->getRecord(ss.str()));
+    }
+
+    // In version 4+, the tensor root_key in bytecode will include the parent
+    // path. The example tensor format is: torch._utils._rebuild_tensor_v2(
```

Comment:
What is this string "torch._utils._rebuild_tensor_v2"?

Also should archive name be "constants/17" and not "bytecode/17" in the comment?

Reply:
[Looks like they are part of the tensor properties](https://github.com/pytorch/pytorch/blob/master/torch/csrc/jit/serialization/pickler.cpp#L363-L364), but I'm not sure what it actually means..

And good catch! It should be `constants/17`